### PR TITLE
Sign Up: Social login first mobile only experiment 

### PR DESCRIPTION
**Pull request by enejb**
#### Changes proposed in this Pull Request

More experiement details in pbxNRc-1a4-p2
Proposal: p58i-bsj-p2

<table>
<tr>
<td>

<div>Default:</div>

Screen Shot 2022-02-16 at 4 49 43 PM

</td>
<td>
<div>Treatment:</div>
Screen Shot 2022-02-17 at 12 31 14 PM

</td>

</tr>
</table>

#### Testing instructions
Point the public-api.wordpress.com to point to your sandbox. (this makes the assignment happen while the experiment is in staging) 

Set the annon user to treatment. 
Notice that you are able to sign up via the flow. 

Start again
Set the annon user to the control flow.
Notice that you are able to sign up via the flow as expected.

### CODE REVIEW

File: client/signup/steps/user/index.jsx
Comment (a8ci18n) [inline client/signup/steps/user/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Create account.' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/user-registration-mobile-social-links)

File: client/signup/steps/user/index.jsx
Comment (a8ci18n) [inline client/signup/steps/user/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Create account.' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/user-registration-mobile-social-links)

File: client/signup/steps/user/index.jsx
Comment (niranjan-uma-shankar) [inline client/signup/steps/user/index.jsx]: Should the `true` boolean be changed here?

**Review by emilylaguna — COMMENTED** [review]
Overall this worked well for me, I tested:
- Treatment: Verified all the buttons and links work correctly
- Treatment: Verified the layout looks good on iOS, unfortunately my Android won't let me connect to the proxy so I wasn't able to test there.
- I did test the Control which is working well.

Looks good to me with the exception of the comment that Niranjan made about the isEligible check

File: client/signup/steps/user/index.jsx
Comment (enejb) [reply client/signup/steps/user/index.jsx]: Good catch!

**Comment by enejb** [comment]
Thanks for the reviews @emilylaguna  and @niranjan-uma-shankar 

@razvanpapadopol Can you take a look at this PR when you have a moment? And let me know if there are any changes that you would like to see?

**Comment by niranjan-uma-shankar** [comment]
@enejb Does it make sense to mirror control by having only the email field for the treatment version? Otherwise, we might be setting treatment up on a disadvantage, given that we know that removing the username and password fields improves registration and site creation in mobile.

File: client/blocks/signup-form/simpler-mobile-form.jsx
Comment (niranjan-uma-shankar) [inline client/blocks/signup-form/simpler-mobile-form.jsx]: Per the design, the copy is `Continue with Your Email`. Should we retain that? If we're sticking with this copy,  I suggest we change the capitalization of `email` to `Email`.

File: client/signup/steps/user/index.jsx
Comment (niranjan-uma-shankar) [inline client/signup/steps/user/index.jsx]: Unless there's a reason for this to be `onboarding` flow only, I suggest expanding this to more signup flows so that we can reach significance earlier.

**Comment by enejb** [comment]
> Does it make sense to mirror control by having only the email field for the treatment version?  

Good point. After chatting with @yaelirub and @emilylaguna. We decided to place the new email form after the buttons as a test. 

The new test looks like this
<table>
<tr>
<td>

<div>Default:</div>


Screen Shot 2022-02-16 at 4 49 43 PM


</td>
<td>
<div>Treatment:</div>
Screen Shot 2022-02-16 at 4 46 28 PM

</td>
</tr>
</table>

Can you take another look?

File: client/signup/steps/user/index.jsx
Comment (niranjan-uma-shankar) [inline client/signup/steps/user/index.jsx]: Looks like this function is no longer needed.

File: client/signup/steps/user/index.jsx
Comment (niranjan-uma-shankar) [inline client/signup/steps/user/index.jsx]: I think we can remove this check since /email is no longer used in the route.

**Review by niranjan-uma-shankar — APPROVED** [review]
A couple of points:

1. In `/with-theme` flow, the ToS text is hidden behind the "Back" button:
Screenshot 2022-02-17 at 10 20 44 AM

2. This is just my opinion, so feel free to ignore, but in the new design, I'd prefer to keep variables to a minimum by not introducing any copy changes. In `treatment`, the subheader copy is different, so any difference in behaviour could also be attributed to the copy in addition to the social buttons. Unless the copy change is crucial for the design change or supports the new design, we probably don't need to make it. 


Since the comments I've left are minor, I'm approving the PR and you may deploy it after making the changes you see fit without waiting for another review.

**Review by razvanpapadopol — COMMENTED** [review]
Hi, @enejb and thanks for the ping!

The usage of ExPlat helper looks good to me. Since it's a signup experiment, I'd recommend also prefetching the assignment in the controller [as we're already doing for other experiments](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/controller.js#L340-L347). When doing that, don't forget to apply the same condition used when evaluating the assignment (`isMobile() && 'wpcc' !== this.props.flowName`)

Also, I agree with @niranjan-uma-shankar on trying to keep the modified variables to a minimum for the experiment to be meaningful.

File: client/signup/steps/user/index.jsx
Comment (enejb) [reply client/signup/steps/user/index.jsx]: Good catch!

File: client/signup/steps/user/index.jsx
Comment (enejb) [reply client/signup/steps/user/index.jsx]: :+1:

**Review by razvanpapadopol — APPROVED** [review]
✅

**Pull request merged**
